client/src/AddFriend.js

```diff
@@ -39,29 +40,31 @@ class AddFriend extends React.Component {
       })
   }
 
+  // TODO: prevent page reload on empty input
   handleAdd (e) {
     // check for empty inputs
     this.setState({ isLoading: true })
     if (!this.state.friendFName || !this.state.friendLName || !this.state.friendEmail) {
-      return this.setState({ status: 406 })
+      return this.setState({ alertType: 'emptyField' })
```

Comment:
To avoid bugs related to misspellings, we can add a `constants` file that defines the strings and import them. This way, the IDE will tell you if you misspelled the variable name.

```
// src/constants.js
module.exports = {
  alertType: {
    EMPTY_FIELD: 'EMPTY_FIELD',
    CREATED: 'CREATED',
    <etc>
}

// src/FormAlert.js
import 'alertType' from './constants'
....
this.setState({ alertType: alertType.EMPTY_FIELD })
```

Reply:
Very smart